src/util/account_util.js

```diff
@@ -318,9 +317,9 @@ function validate_assume_role_policy(policy) {
 }
 
 // User name is first part is user provided name, and second part
-// is root account name, This will make the user name uniq accross system.
-function get_account_name_from_username(username, requesting_account_name) {
-    return new SensitiveString(`${username}:${requesting_account_name}`);
+// is root account id, This will make the user name uniq accross system.
+function get_account_name_from_username(username, requesting_account_id) {
+    return new SensitiveString(`${username}:${requesting_account_id}`);
 }
```

Comment:
From the function I understand that `requesting_account_id` should be string, but from the use I saw that you passed `ObjectId` type.